File: src/Java/Euclid.java
package Java;

import Java.Test;
import edu.princeton.cs.algs4.StdOut;

public class Euclid {
    public static void main(String[] args) {
        System.out.println("JAVA");
        int result = gcd(1111111, 1234567);
        System.out.println("Result: " + result);
        StdOut.print("Result: " + result);
    }

    public static int gcd(int p, int q) {
        if (q == 0) {
            return p;
        } else {
            int r = p % q;
            System.out.println("New Argument: p = " + q + " q = " + r);
            return gcd(q, r);
        }
    }
}

File: src/Kotlin/Euclid.kt
package Kotlin

fun main(args: Array<String>) {
    println("KOTLIN")
    var result: Int = gcd(1111111, 1234567)
    println("Result: $result")
}

fun gcd(p: Int, q: Int): Int {
    if (q == 0) {
        return p
    } else {
        var r: Int = p % q
        println("New Argument: p = $q q = $r")
        return gcd(q, r)
    }
}